windows-installer/construct.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+import ctypes
+import subprocess
+import os
+import sys
+import shutil
+
+
+def main():
+    print('Gather runtime data')
+
+    try:
+        subprocess.check_output(['choco', '--version'])
+    except subprocess.CalledProcessError:
+        raise RuntimeError('Error: Chocolatey (https://chocolatey.org/) needs'
+                           'to be installed to run this script.')
+
+    script_path = os.path.realpath(__file__)
+    repo_path = os.path.dirname(os.path.dirname(script_path))
+    build_path = os.path.join(repo_path, 'windows-installer', 'build')
+
+    venv_path = os.path.join(build_path, 'venv-config')
+    venv_python = os.path.join(venv_path, 'Scripts', 'python.exe')
+    installer_cfg_path = os.path.join(build_path, 'installer.cfg')
+    wheels_path = os.path.join(build_path, 'wheels')
+
+    certbot_version = subprocess.check_output([sys.executable, '-c', 'import certbot; print(certbot.__version__)'],
+                                              universal_newlines=True, cwd=repo_path).strip()
+
+    certbot_packages = ['acme', '.']
+    certbot_packages.extend([name for name in os.listdir(repo_path) if name.startswith('certbot-dns-')])
+
+    print('Copy assets')
+
+    os.makedirs(build_path, exist_ok=True)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'certbot.ico'), build_path)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'run.py'), build_path)
+
+    print('Prepare pynsist config')
+
+    with open(os.path.join(installer_cfg_path), 'w') as file_h:
+        file_h.write("""\
+[Application]
+name=Certbot
+version={certbot_version}
+icon=certbot.ico
+publisher=Electronic Frontier Fundation
+script=run.py
```

Comment:
What do you think about deleting this start menu entry entirely based on the advice at https://docs.microsoft.com/en-us/windows/win32/shell/desktop-app-tiles-on-the-start-screen#decide_the_right_entry_points_to_include_in_the_start_screen which is linked from https://pynsist.readthedocs.io/en/latest/cfgfile.html#shortcut-sections?

If we don't do this for some reason, this results in a start menu entry that says
```
Certbot
App
```
but it's not the Certbot app, it's the docs.

Do you think there's a way we can clean this up a bit? I didn't test it, but it seems like we could be able to remove this and create a [shortcut](https://pynsist.readthedocs.io/en/latest/cfgfile.html#shortcut-sections) with a more descriptive name.

Reply:
Since certbot is a pure command line program, I initially wanted to not put any shortcut in the Start Menu. But you can't with the way pynsist is designed, since it requires a `entrypoint`, a `script` or a `target` in Application section to create a shortcut with.

The two firsts keywords are python centric. The last one is any acceptable target for a Windows shortcut. 

I agree opening a browser on the Certbot online help is not great. 

So I did another approach. The most useful action from Start Menu would be to open a command line, since it is how you interact with Certbot. So I created a batch script that do that an set it as the target for the shortcut. This script ensures that we have the administrative privileges, since it is necessary for Certbot, open the UAC if needed, then start the cmd on this elevated privileges with a little welcome echo. 

So far I think it is the most suitable thing to do.